include/acquire.zarr.h

```diff
@@ -0,0 +1,274 @@
+                                            const ZarrS3Settings* s3_settings);
+
+    /**
+     * @brief Set the data type, compressor, codec, compression_settings level,
+     * and shuffle for the Zarr stream.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param[in] compression_settings The compression_settings settings.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_set_compression(
+      ZarrStreamSettings* settings,
+      const ZarrCompressionSettings* compression_settings);
+
+    /**
+     * @brief Set the data type for the Zarr stream.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param[in] data_type The data type.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_set_data_type(ZarrStreamSettings* settings,
+                                                ZarrDataType data_type);
+
+    /**
+     * @brief Reserve space for dimensions in the Zarr stream settings struct.
+     * @detail *Must* precede calls to ZarrStreamSettings_set_dimension. We
+     * require at least 3 dimensions to validate settings, but you may set up to
+     * 32 dimensions.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param[in] count The number of dimensions to reserve space for.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_reserve_dimensions(
+      ZarrStreamSettings* settings,
+      size_t count);
+
+    /**
+     * @brief Set properties for an acquisition dimension.
+     * @detail The order of the dimensions in the Zarr stream is the order in
+     * which they are set. The first dimension set is the slowest varying
+     * dimension, and the last dimension set is the fastest varying dimension.
+     * For example, if the dimensions are set in the order z, y, x, the fastest
+     * varying dimension is x, the next fastest varying dimension is y, and the
+     * slowest varying dimension is z.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param[in] index The index of the dimension to set. Must be less than the
+     * number of dimensions reserved with ZarrStreamSettings_reserve_dimensions.
+     * @param[in] dimension The dimension's settings.
+     */
+    ZarrStatus ZarrStreamSettings_set_dimension(
+      ZarrStreamSettings* settings,
+      size_t index,
+      const ZarrDimensionProperties* dimension);
+
+    /**
+     * @brief Set the multiscale flag for the Zarr stream.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param[in] multiscale A flag indicating whether to stream to multiple
+     * levels of detail.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_set_multiscale(ZarrStreamSettings* settings,
```

Comment:
The most significant design change we should consider is the management of the settings object ownership. Instead of the API owning this object (allocating space, providing copy and dispose functions, etc.), we should have users instantiate and configure the fields themselves. This approach would eliminate the need for object lifetime management, and mutator and accessor functions. When creating a stream, users would pass a pointer to the settings object (and potentially a pointer to the store object), and the API would internally validate the settings and populate defaults as needed.